sdk/eventhub/event-hubs/src/util/timerLoop.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * A timer loop is a loop over multiple instances of a promise every specific
+ * time interval. It is different from `setInterval` in that it waits until the
+ * promise is settled before it goes to the next iteration.
+ * */
+export interface TimerLoop {
+  /**
+   * Starts the timer loop. If the loop is already running, the loop will be
+   * restarted.
+   */
+  start: () => void;
+  /**
+   * Stops the timer loop. If the loop is not running, it is a no-op.
+   */
+  stop: () => void;
+}
+
+/**
+ * Creates a timer loop with the given timeout and task.
+ * @internal
+ */
+export function createTimerLoop(
+  timeoutInMs: number,
+  createTask: () => Promise<unknown>
+): TimerLoop {
+  let token: ReturnType<typeof setTimeout>;
+  const loop = {
+    start: () => {
+      clearTimeout(token);
+      token = setTimeout(() => createTask().then(loop.start), timeoutInMs);
```

Comment:
if `createTask` rejects then the loop will stop, should we do anything to log or handle this?

Reply:
Addressed in https://github.com/Azure/azure-sdk-for-js/pull/25214/commits/9e1faf1551c63799b5ad35689ebb081699067fa1.

Reply:
I give this more thinking and talked to @jsquire about it. Authentication errors are retryable and if they keep failing, the link will probably get closed. I updated the loop abstraction to eat up any unhandled errors and keep looping. I also added a test case.